model_analyzer/record/gpu_free_memory.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import time
+from model_analyzer.record.gpu_record import GPURecord
+
+
+class GPUFreeMemory(GPURecord):
+    """
+    The free memory in the GPU.
+    """
+    def __init__(self, device, free_mem):
+        self._free_mem = free_mem
+        self._device = device
+        self._timestamp = time.time()
```

Comment:
Can you please also add the ability to override this value? For example, having a time parameter that allows you to override the default value which is the current time.

Reply:
Yes. I think I can do that. Should that be for all records in general? Or only GPURecords?

Reply:
I think it should be for all the records. The monitoring library may provide the time for when the record was measured. I think DCGM provided that and that is the primary reason I am asking for this. Thanks.

Reply:
Okay, I will add a method to the super class `update_time` that just sets it to the current time. Or do you want this to act like a sequence number? Maybe we can fix this in a separate PR closer to the use case?

Reply:
This should be a relative time to the start of the test, not a date time stamp. This is how profilers like nsight systems work.